File: ros_ws/code/turtle_twist_publisher.py
#!/usr/bin/env python3
# use the default python compiler

# import ROS python library
import rospy

#import the Twist message type
from geometry_msgs.msg import Twist

    
# initialize a node named 'twist_publisher'
rospy.init_node("twist_publisher")

# creates a publisher for the /turtle1/cmd_vel topic
pub=rospy.Publisher('/turtle1/cmd_vel',Twist,queue_size=10)

# adjust the publishing rate to be 1 Hz
rate=rospy.Rate(1)

# loop forever or until interrupted by CTRL+C
while not rospy.is_shutdown():
    # Create a message and set the values of the x component of the linear velocity
    # and the z component of the angular velocity. Other fields will remain zero.
    twist_msg = Twist()
    twist_msg.linear.x=1.0
    twist_msg.angular.z=0.8
    
    # Use the publisher handler (pub) to publish the message
    pub.publish(twist_msg)
    
    #wait until the rate is satisfied (1Hz)
    rate.sleep()

File: ros_ws/code/turtle_twist_publisher.cpp
// Include the necessary ROS header file for C++ programming
#include <ros/ros.h>

// Include the header file for the Twist message type
#include <geometry_msgs/Twist.h>

int main(int argc, char** argv){
	//initialize a node named 'twist_publisher'
	ros::init(argc,argv,"twist_publisher");
	
	//create a nodeHandle object to interface with the ROS network.
	ros::NodeHandle nh;

	// creates a publisher to send messages of type geometry_msgs::Twist 
	// on the "/turtle1/cmd_vel" topic with a queue size of 10
	ros::Publisher pub= nh.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel",10);

	// publishes messages at rate of 1 times per second
	ros::Rate rate(1);
	
	// while the node is running
	while(ros::ok()) {
		
		// Create a Twist message with linear and angular velocities
		geometry_msgs::Twist twist_msg;
		twist_msg.linear.x=1.0;
		twist_msg.angular.z=0.8;
		
		// publish tje linear and angular velocities
		pub.publish(twist_msg);
		
		//delay to achieve the desired publishing rate
		rate.sleep();
	}
	
	return 0;
}

